compat/signal.cc

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (C) 1996-2025 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#include "squid.h"
+#include "compat/signal.h"
+
+#if _SQUID_WINDOWS_ || _SQUID_MINGW_
+
+#ifdef HAVE_PSAPI_H
+#include <psapi.h>
+#endif
+
+static void
+GetProcessName(pid_t pid, char *ProcessName)
+{
+    strcpy(ProcessName, "unknown");
```

Comment:
If this code survives, please return a nil name instead of using a special value to mean "we do not know what the process name is [because some system calls have failed]" (that callers do not know about!).

N.B. I am aware that official code uses this hack.